src/pykeen/models/baseline/utils.py

```diff
@@ -45,17 +45,17 @@ def marginal_score(
 
     e, r = entity_relation_batch.cpu().numpy().T
 
-    if per_entity is None and per_relation is None:
-        raise NotImplementedError
-    elif per_entity is None:  # and per_relation is not None
+    if per_relation is not None and per_entity is None:
         scores = per_relation[r]
-    elif per_relation is None:  # and per_entity is not None
+    elif per_relation is None and per_entity is not None:
         scores = per_entity[e]
-    else:  # per_relation is not None and per_entity is not None
+    elif per_relation is not None and per_entity is not None:
         e_score = per_entity[e]
         r_score = per_relation[r]
         scores = e_score.multiply(r_score)
         scores = sklearn_normalize(scores, norm="l1", axis=1)
+    else:
+        raise AssertionError  # for mypy
```

Comment:
@cthoyt if you have a nicer solution, please go ahead :slightly_smiling_face:

Reply:
Nope this is good enough